calico_containers/pycalico/tests/test_util.py

```diff
@@ -0,0 +1,116 @@
+    inet6 2620:104:4008:69:8d7c:499f:2f04:9e55/64 scope global temporary dynamic 
+       valid_lft 603690sec preferred_lft 84690sec
+    inet6 2620:104:4008:69:a00:27ff:fe73:c8d0/64 scope global dynamic 
+       valid_lft 604698sec preferred_lft 86298sec
+    inet6 fe80::a00:27ff:fe73:c8d0/64 scope link 
+       valid_lft forever preferred_lft forever
+3: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
+    link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff
+    inet 172.17.42.1/24 scope global docker0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::188f:d6ff:fe1f:1482/64 scope link 
+       valid_lft forever preferred_lft forever
+"""
+
+MOCK_IP_ADDR_LOOPBACK = \
+"""
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 
+    inet6 ::1/128 scope host 
+       valid_lft forever preferred_lft forever
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+"""
+
+
+class TestUtil(unittest.TestCase):
+
+  @patch("pycalico.util.check_output", autospec=True)
+  def test_get_host_ips_standard(self, m_check_output):
+    # Test IPv4
+    m_check_output.return_value = MOCK_IP_ADDR
+    addrs = get_host_ips(version=4)
+    m_check_output.assert_called_with(["ip", "-4", "addr"])
+    self.assertEquals(addrs, ['172.24.114.18/24', '172.17.42.1/24'])
+
+    # Test IPv6
+    addrs = get_host_ips(version=6)
+    m_check_output.assert_called_with(["ip", "-6", "addr"])
+    self.assertEquals(addrs, ['2620:104:4008:69:8d7c:499f:2f04:9e55/64',
+                  '2620:104:4008:69:a00:27ff:fe73:c8d0/64',
+                  'fe80::a00:27ff:fe73:c8d0/64',
+                  'fe80::188f:d6ff:fe1f:1482/64'])
+
+  @patch("pycalico.util.check_output", autospec=True)
+  def test_get_host_ips_loopback_only(self, m_check_output):
+    # Test Loopback
+    m_check_output.return_value = MOCK_IP_ADDR_LOOPBACK
+    addrs = get_host_ips(version=4)
+    m_check_output.assert_called_with(["ip", "-4", "addr"])
+    self.assertEquals(addrs, [])
+
+    addrs = get_host_ips(version=6)
+    m_check_output.assert_called_with(["ip", "-6", "addr"])
+    self.assertEquals(addrs, [])
+
+  @patch("pycalico.util.check_output", autospec=True)
+  def test_get_host_ips_exclude(self, m_check_output):
+    # Exclude "docker0"
+    m_check_output.return_value = MOCK_IP_ADDR
+    addrs = get_host_ips(version=4, exclude=["docker0"])
+    m_check_output.assert_called_with(["ip", "-4", "addr"])
+    self.assertEquals(addrs, ['172.24.114.18/24'])
+
+    addrs = get_host_ips(version=6, exclude=["docker0"])
+    m_check_output.assert_called_with(["ip", "-6", "addr"])
+    self.assertEquals(addrs, ['2620:104:4008:69:8d7c:499f:2f04:9e55/64',
+                  '2620:104:4008:69:a00:27ff:fe73:c8d0/64',
+                  'fe80::a00:27ff:fe73:c8d0/64'])
+
+    # Exclude empty list
+    addrs = get_host_ips(version=4, exclude=[""])
+    m_check_output.assert_called_with(["ip", "-4", "addr"])
```

Comment:
Since you are calling get_host_ips multiple times here, this assert is not really meaningful (since we've already asserted that m_check_output is called with this argument).

A better pattern to use when checking calls is the following:

```
m_check_output.assert_called_once_with( ... )
m_check_output.reset_mock()
```
This asserts that the mock was called with the correct args once and only once (if it is called twice, it will fail).  reset_mock() resets the call counters on the mock so that subsequent assert_called_once_with() calls will only see the latest call.

Another way to do this would be to use assert_has_calls(), which allows you to specify the complete list of calls.